### PR TITLE
Security with autorisation and autentication  

### PR DESCRIPTION
**Pull request by Sofasmile**

### CODE REVIEW

File: src/main/java/net/spring/service/RoleService.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package net.spring.service;
+
+import net.spring.entity.user.Role;
+
```
Comment (Taniann) [inline src/main/java/net/spring/service/RoleService.java:4]: remove redundant blank line

File: src/main/java/net/spring/service/UserService.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package net.spring.service;
+
+import net.spring.entity.user.User;
+
+public interface UserService {
+    User findByUserName(String userName);
```
Comment (Taniann) [inline src/main/java/net/spring/service/UserService.java]: better name findByUserName -> findByName

you already have word "User" in service name

File: src/main/java/net/spring/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+    void saveUser(User user);
```
Comment (Taniann) [inline src/main/java/net/spring/service/UserService.java]: the same saveUser -> save

File: src/main/java/net/spring/service/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final RoleService roleService;
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
+    private static final String ROLE = "USER";
+
+    @Override
+    public User findByUserName(String userName) {
+        return userRepository.findByUserName(userName).orElse(null);
```
Comment (Taniann) [inline src/main/java/net/spring/service/UserServiceImpl.java]: or else better throw custom exception with message for example "User with this username not found"

**Review by Taniann — CHANGES_REQUESTED** [review]
good but please make a little fixing

File: src/main/java/net/spring/controller/LoginController.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    public ModelAndView registration() {
+        ModelAndView modelAndView = new ModelAndView();
+        User user = new User();
+        modelAndView.addObject("user", user)
+                .setViewName("registration");
+        return modelAndView;
+    }
+
+    @PostMapping("/registration")
+    public ModelAndView createNewUser(User user, BindingResult bindingResult) {
```
Comment (asulyma) [inline src/main/java/net/spring/controller/LoginController.java]: Please add NotNull annotation

File: src/main/java/net/spring/controller/AppController.java [hunk 6]
```diff
@@ -21,37 +20,37 @@
...
-    @GetMapping("/new")
+    @GetMapping("/students/new")
     public String createNewStudentForm(@Valid Model model) {
         Student student = new Student();
         model.addAttribute("student", student);
         return "new_student";
     }
 
-    @PostMapping("/save")
+    @PostMapping("/students/save")
```
Comment (asulyma) [inline src/main/java/net/spring/controller/AppController.java]: Please remove "students" from this and add it to @RequestMapping

File: src/main/java/net/spring/entity/user/User.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    private Long id;
+
+    @Column(name = "name")
+    private String userName;
+
+    @Column(name = "password")
+    private String password;
+
+    @Column(name = "active")
+    private int active;
```
Comment (asulyma) [inline src/main/java/net/spring/entity/user/User.java]: change on boolean

**Review by Taniann — APPROVED** [review]
good

**Pull request merged**